docs/design/maven.md

```diff
@@ -0,0 +1,130 @@
+# Generic artifact fetching
+
+## Introduction
+
+This document will describe high-level implementation overview for supporting fetching maven artifacts in cachi2. This
+proposal depends on generic artifact fetching, which is described in this [design PR][generic-pr].
+
+## Context
+
+[Maven] is a Java ecosystem build system and functions basically as a package manager, among other things. Maven provides a
+standardized way to specify dependencies in a `pom.xml` file, which is then used to fetch dependencies from a maven repository.
+This feature in cachi2 isn't concerned with pre-fetching those, but rather to provide a way to fetch individual artifacts
+from maven repository. This is useful for cases where the build only needs specific artifacts from big projects, saving
```

Comment:
Do you mean that instead of pre-fetching entire projects the plan is to pre-fetch just some parts of these already built projects? I am asking as someone with a very limited maven experience.

Reply:
Yes, that's correct. This is analogous to how OSBS works, here's how they used to [specify](https://osbs.readthedocs.io/en/osbs_ocp3/users.html?highlight=fetch#fetch-artifacts-pnc-yaml) the artifacts to fetch. This feature is similar, but much more generic, and reusable outside of one specific usecase.

Reply:
Thank you. Then I suggest replacing "those" with "entire maven projects" for clarity.

Reply:
Replaced.